pages/pipelines/hosted_agents/code_access.md

```diff
@@ -18,3 +16,44 @@ To use a private GitHub repository with Buildkite hosted agents, you'll need to
 ## Public repositories
 
 Buildkite does not require any special permissions to access public repositories.
+
+## Other private repositories
+
+There are 2 key aspects to utilizing a private repository with Buildkite hosted agents.
+
+1. Adding an SSH key as a Secret to the Buildkite hosted agent cluster.
+1. Adding the plugin to the initial pipeline steps, and any further steps within the pipeline that are loaded.
+
+### Adding the SSH key secret
+
+Navigate to **Agents** from the top menu, and open the **Cluster** for Buildkite hosted agents. In the left-hand side navigation, there will be a **Secrets** option to follow. Clicking the **New Secret** button will open a modal to capture the new secret.
+
+<%= image "secret-creation.png", width: 1516, height: 478, alt: "Creating a secret called GIT_SSH_CHECKOUT_PLUGIN_SSH_KEY" %>
+
+This secret should contain the full private key (including the header and footer) that will be used to access the repository.
+
+If there are multiple distinct keys to be used throughout the cluster, make sure to name them appropriately so they can each be used at their correct times.
+
+### Adding a new pipeline
+
+With the secret available, a new pipeline can be added that will use it and allow for the Git repository to be used.
```

Comment:
Some optional phrasing suggestions


With the secret now available, you can add a new pipeline to use it and access the Git repository.

The availability of the secret allows the creation of a new pipeline to utilize it and access the Git repository.

Once the secret is available, a new pipeline can be set up to use it and enable Git repository access.